sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -869,6 +874,8 @@ case class AdaptiveExecutionContext(session: SparkSession, qe: QueryExecution) {
    */
   val stageCache: TrieMap[SparkPlan, ExchangeQueryStageExec] =
     new TrieMap[SparkPlan, ExchangeQueryStageExec]()
+
+  val shuffleIds: TrieMap[Int, Boolean] = new TrieMap[Int, Boolean]()
```

Comment:
what does the value mean? BTW, `stageCache` uses `TrieMap` because the key is `SparkPlan`. For int key, I think normal hash map works fine

Reply:
I think a concurrent hash map is still required since the context are shared between the main query and all sub queries?

Reply:
yea, concurrent hash map with int key should be good here.

Reply:
Updated